src/PerimeterxContext.php

```diff
@@ -68,6 +62,27 @@ public function __construct($pxConfig)
         $this->sensitive_route = $this->checkSensitiveRoutePrefix($pxConfig['sensitive_routes'], $this->uri);
     }
 
+    private function extractIP($pxConfig) {
+        $pxConfig['logger']->warning('looking for ip');
+        $all_headers = getallheaders();
+
+        if (isSet($pxConfig['ip_headers'])) {
+            foreach ($pxConfig['ip_headers'] as $header) {
+                if (isSet($all_headers[$header])) {
+                    return $all_headers[$header];
+                }
+            }
+        }
+        if (isset($pxConfig['custom_user_ip'])) {
+            return $pxConfig['custom_user_ip']($this);
+        }
+        if (function_exists('pxCustomUserIP')) {
```

Comment:
`function_exists`, where should this function exist? 
shouldn't this function be extracted from `$pxConfig`?

and also, there is no documentation for this, is it setting the ip variable in the `$ctx`?